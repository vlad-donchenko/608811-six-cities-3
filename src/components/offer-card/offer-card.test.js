import React from "react";
import renderer from "react-test-renderer";
import OfferCad from "./offer-cad.jsx";

const offer = {
  name: `Place 2`,
  index: 0,
  image: `apartment-02.jpg`,
  price: 400,
  type: `Private room`,
  rating: 4,
  isBookmark: true,
  isPremium: false,
};

it(`Render OfferCad`, () => {
  const tree = renderer.create(<OfferCad offer={offer} onOfferHover={() => {}} onTitleClick={() => {}}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
