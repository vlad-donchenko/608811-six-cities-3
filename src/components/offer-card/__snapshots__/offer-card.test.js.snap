// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <OfferCad/> 1`] = `
<article
  className="cities__place-card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
  <div
    className="cities__place-card__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Beautiful & luxurious studio at great location - 1"
        className="place-card__image"
        height="200"
        src="img/apartment-02.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button "
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "96%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Beautiful & luxurious studio at great location - 1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      apartment
    </p>
  </div>
</article>
`;
